deepchecks/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,234 @@
+    """
+        Calculate drift between train and test datasets as a whole, using a model trained to distinguish between them.
+
+        Check trains a model to distinguish between train and test datasets, called a domain classifier. Then,
+        it calculates the feature importance for the domain classifier model. The result of the check is based on the
+        AUC of the domain classifier model, and the check displays the change in distribution between train and test
+        for the top features according to the calculated feature importance.
+
+        Args:
+            n_top_features (int):
+                Amount of columns to show ordered by domain classifier feature importance.
+            min_feature_importance (float): Minimum feature importance to show in the check display.
+            max_num_categories (int):
+                Only for categorical columns. Max number of allowed categories. If there are more,
+                they are binned into an "Other" category in the display. If max_num_categories=None, there is no limit.
+            sample_size (int):
+                Amount of rows to use from each dataset for the training of the domain classifier.
+            random_state (int):
+                Random seed for the check.
+        """
+
+    def __init__(
+        self,
+        n_top_features: int = 3,
+        min_feature_importance: float = 0.05,
+        max_num_categories: int = 10,
+        sample_size: int = 10000,
+        random_state: int = 0,
+    ):
+        super().__init__()
+
+        self._cat_features = None
+        self.n_top_features = n_top_features
+        self.min_feature_importance = min_feature_importance
+        self.max_num_categories = max_num_categories
+        self.sample_size = sample_size
+        self.random_state = random_state
+
+    def run(self, train_dataset, test_dataset, model=None) -> CheckResult:
+        """Run check.
+
+        Args:
+            train_dataset (Dataset): The training dataset object. Must contain a label column.
+            test_dataset (Dataset): The test dataset object. Must contain a label column.
+            model: not used in this check.
+
+        Returns:
+            CheckResult:
+                value: dictionary containing the domain classifier auc and a dict of column name to its feature
+                       importance as calculated for the domain classifier model.
+                display: distribution graph for each column for the columns most explaining the dataset difference,
+                         comparing the train and test distributions.
+
+        Raises:
+            DeepchecksValueError: If the object is not a Dataset or DataFrame instance
+        """
+
+        train_dataset = Dataset.validate_dataset_or_dataframe(train_dataset)
+        test_dataset = Dataset.validate_dataset_or_dataframe(test_dataset)
+
+        features = train_dataset.validate_shared_features(test_dataset, self.__class__.__name__)
```

Comment:
I think Matan updated the name of the class as an attribute of self, so you can use that instead of self.__class__.__name__. I think